cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -870,6 +879,7 @@ public ScheduleDetail apply(@Nullable JsonElement input) {
     ProgramType programType = ProgramType.valueOfSchedulableType(scheduleFromRequest.getProgram().getProgramType());
     String programName = scheduleFromRequest.getProgram().getProgramName();
     ProgramId programId = applicationId.program(programType, programName);
+    authorizationEnforcer.enforce(programId, authenticationContext.getPrincipal(), Action.EXECUTE);
```

Comment:
If we only allow `EXCECUTE` privilege to add/remove schedules then admins will not be able to deploy applications with schedules, right?